### ISSUE TITLE
Unfollow a node

### ISSUE THREAD

**Issue by polgiron** [Iotic-Labs/demo#3]
It could be interesting to have a unfollow function.
For exemple, a node is following another one but at one point the second node is turned off, the first node should notice this change and unfollow it.

**Comment by whaleygeek** [comment]
Yes, this is a good idea, and the sandbox that I put together does have the concept of an unfollow() although it is not yet implemented.

One issue we will have to work through, is how a Node/Point code would hear about the unfollow - I understand from the dev team that bind functionality is supposed to be policy based (i.e. an access control list with a set of rules to decide when to allow a bind), and as such notifcations of bind and unbind would not normally be transferred all the way to the end device.

I'll talk this through with the dev team and report back here when I've got a clearer view of how this might work.

**Comment by whaleygeek** [comment]
I think this issue is less about the unfollow() semantics, but more about the potential of "isAlive" semantics. 

We've decided that this is not going to be in the kernel part of the underlying infrastructure, but implemented as a higher level service in the UserAgent. The UserAgent is the python API that you call directly, and it all runs on your computer.

Thus, I think how this might work is as follows:

1. the UserAgent, on your behalf, creates a hidden system feed, perhaps "sys$health" for all nodes, which is then advertised automatically. So, in your example, your second node would have a sys$health feed automatically advertised when you all the createNode().

2. wakeup() and sleep() will cause messages shared via this sys$health feed, as "AWAKE" and "SLEEP" payloads.

3. a 1 second heartbeat message will be shared on this feed "HEARTBEAT" whenever the Node code is running.

4. In your example, the first node (the node that did the following) is interested in whether the followed node is still there or not. So, your app code could find() the "sys$health" feed, follow() it, and route it to a callback that receives these state messages. Your app will then know when the node is turned on (AWAKE), turned off in a controlled way (SLEEP), and get heartbeats when it is awake (HEARTBEAT). If a heartbeat is not received in a few seconds, you could assume that either there is a lot of network congestion, or that the node is no longer powered on, and unfollow() it.

5. It's possible that I might wrap up this receiving health monitor in a UserAgent service API so it all happens automatically, including checking timestamps and deciding if the heartbeats have disappeared for long enough to indicate the device is no longer there. That way, you could just ask the UserAgent API to call a callback on state changes, and the whole thing would feel very event driven in your app.

6. We might be able to find a way to make the heartbeats more efficient, so that if a Node.sys$health has no follows, it does not send the heartbeat in the first place. I think we know from the infrastructure how many follows a Node has, so this might be possible in future editions.

What I would say is that, you can implement this design pattern yourself today, without any additions to the UserAgent - i.e. you can prototype the idea of a sys$health feed from a Node, in your app, and report back if it solves the problem you are trying to deal with at the moment, and we can use your feedback on this activity to inform the design of this service inside the UserAgent.

**Comment by whaleygeek** [comment]
I also note that unfollow() (i.e. unbind()) needs to be implemented in the UserAgent, in order for you to complete your overall application task here, regardless of how heart-beating and health state is indicated. I will write the unbind() logic in the next minor release of the UserAgent API.
